server/src/main/java/org/opensearch/cluster/metadata/Metadata.java

```diff
@@ -844,6 +844,12 @@ public Map<String, View> views() {
         return Optional.ofNullable((ViewMetadata) this.custom(ViewMetadata.TYPE)).map(ViewMetadata::views).orElse(Collections.emptyMap());
     }
 
+    public Map<String, QueryGroup> queryGroups() {
+        return Optional.ofNullable((QueryGroupMetadata) this.custom(QueryGroupMetadata.TYPE))
+            .map(QueryGroupMetadata::queryGroups)
+            .orElse(Collections.emptyMap());
+    }
```

Comment:
Missing merge from mainline? https://github.com/opensearch-project/OpenSearch/pull/13669/ already has this method?

Reply:
We can reuse the existing method, we just need to change the access modifier